### ISSUE TITLE
OptionalWhenBraces false negative for nested when wrapped in braces

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#4245]
Using 1.18.1

The following code does not fail for OptionalWhenBraces:

```
fun main() {
  when(Random.nextBoolean()) {
    true -> {
      when(Random.nextBoolean()) {
        true -> {
          println("true")
        }
        false -> {
          println("false")
        }
      }

      println("end")
    }
  }
}
```

_edited to simplify repro_

**Comment by eygraber** [comment]
@BraisGabin looks like there's either been a regression or another case that wasn't handled:

```
import android.icu.text.PluralRules
import kotlinx.serialization.json.JsonObject
import java.util.Locale

fun JsonObject.plural(
  plural: Number?,
  language: String = "en"
) = when(plural) {
  null -> get("other")

  else -> {
    when {
      "other" in this -> get(
        PluralRules.forLocale(
          Locale.forLanguageTag(language)
        ).toString()
      )

      // other must be present even if a present rule would match
      else -> null
    }
  }
}
```

**Comment by BraisGabin** [comment]
Can you open it as a new issue? You could link there this issue. It doesn't seem like a regression because the fix had a test to avoid that regression. But it's a clear false negative. And thanks for the report, the false negatives are difficult to spot :)
